distributed/tests/test_worker.py

```diff
@@ -1778,6 +1779,55 @@ async def test_story(c, s, w):
     assert w.story(ts) == w.story(ts.key)
 
 
+def donothing():
+    import time
+
+    from distributed._concurrent_futures_thread import WorkerThreadInterrupt
+
+    try:
+        for _ in range(100):
+            time.sleep(0.01)
+        raise KeyboardInterrupt
+    except WorkerThreadInterrupt:
+        get_worker().attr = "Quitting"
+
+
+@gen_cluster(client=True, nthreads=[("127.0.0.1", 1)], timeout=2)
+async def test_interrupt(c, s, w):
+    import concurrent.futures
+
+    w.interruptor = True
+    fut0 = c.submit(donothing)
+    await async_wait_for(lambda: bool(w.active_threads), 1)
+    fut1 = c.submit(lambda: True)  # blocked
+    assert not fut1.done()
```

Comment:
I believe you should insert a small sleep here `await asyncio.sleep(0.01)` since even if this function returns almost immediately, the future cannot be done in the same loop cycle as it has been submitted